### PR TITLE
schema: add json for modules K-L

### PR DESCRIPTION
**Pull request by blackboxsw**
## Proposed Commit Message

```
schema: add JSON defs for modules K-L

Includes:
- cc_keyboard: migrated legacy scheme to cloud-init-schema.json
- cc_keys_to_console:
- cc_landscape: added schema defs for most frequent client keys
- cc_locale
- cc_lxd

Dropping duplicates docs as schema examples now cover them
 - doc/examples/cloud-config-landscape.txt 
 - doc/examples/cloud-config-lxd.txt 

LP: #1858899, #1858900, #1858901, #1858902
```

## Test Steps

```bash
tox -e py3 tests/unittests/config
tox -e doc
xdg-open doc/rtd_html/topics/modules.html
for mod in cc_locale cc_lxd cc_landscape cc_keyboard cc_keys_to_console; do
    PYTHONPATH=. python3 -m cloudinit.cmd.main devel schema --docs \$mod;
done
```
PYTHONPATH=. python3 -m cloudinit.cmd.main devel schema --docs cc_landscape

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

File: cloudinit/config/cc_lxd.py [hunk 1]
```diff
@@ -4,61 +4,79 @@
...
 
 _DEFAULT_NETWORK_NAME = "lxdbr0"
 
 
+MODULE_DESCRIPTION = """\
+This module configures lxd with user specified options using ``lxd init``.
+If lxd is not present on the system but lxd configuration is provided, then
+lxd will be installed. If the selected storage backend is zfs, then zfs will
+be installed if missing. If network bridge configuration is provided, then
+lxd-bridge will be configured accordingly. Configure the system locale and
```
Comment (TheRealFalcon) [inline cloudinit/config/cc_lxd.py:30]: Copypasta

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Overall looks good. I left some comments inline. Additionally,

For landscape, we might want a way to indicate that the schema params are optional. Doesn't necessarily inline in every parameter, but it's probably worth mentioning in the module description.

General comment: When setting `distros` or `frequency` in the meta schema, I don't think we need a separate variable definition if that's the only place it's used. E.g., use `"distros": ["all"],` and `"frequency": PER_INSTANCE,` instead of using new variables that don't get used anywhere else.

Rebase will be needed too.

File: cloudinit/config/cloud-init-schema.json [hunk 2]
```diff
@@ -543,6 +543,243 @@
...
+          },
+          "required": ["layout"],
+          "additionalProperties": false
+        }
+      }
+    },
+    "cc_keys_to_console": {
+      "type": "object",
+      "properties": {
+        "ssh": {
```
[the remark this answers is not in the document]
Comment (blackboxsw) [reply cloudinit/config/cloud-init-schema.json:579]: I would like to do this too and yes it's the beginning of an attempt to move out of the shared top-level scope key namespace.

[the remark this answers is not in the document]
Comment (blackboxsw) [reply cloudinit/config/cloud-init-schema.json:579]: Added https://warthogs.atlassian.net/browse/SC-885 for discussion. maybe it's a good idea. maybe not.

File: cloudinit/config/cloud-init-schema.json [hunk 3]
```diff
@@ -543,6 +543,243 @@
...
+                  "type": "string",
+                  "description": "The account this computer belongs to."
+                },
+                "registration_key": {
+                  "type": "string",
+                  "description": "The account-wide key used for registering clients."
+                },
+                "tags": {
+                  "type": "string",
+                  "pattern": "^[-_0-9a-zA-Z]+(,[-_0-9a-zA-Z]+)+$",
```
[the remark this answers is not in the document]
Comment (blackboxsw) [reply cloudinit/config/cloud-init-schema.json:650]: +1

**Comment by blackboxsw** [comment]
> General comment: When setting `distros` or `frequency` in the meta schema, I don't think we need a separate variable definition if that's the only place it's used. E.g., use `"distros": ["all"],` and `"frequency": PER_INSTANCE,` instead of using new variables that don't get used anywhere else.


Turns out we do need them (see CI failures :)) because cloudinit.stages and [cloudinit.config.fixup_module](https://github.com/canonical/cloud-init/blob/main/cloudinit/config/__init__.py#L33-L44) looks for [module.frequency](https://github.com/canonical/cloud-init/blob/main/cloudinit/stages.py#L1063) and [module.distros](https://github.com/canonical/cloud-init/blob/main/cloudinit/stages.py#L1122) attributes directly, which I think we need to fix to instead look for module.meta[]. I've added a separate [jira card #888](https://warthogs.atlassian.net/browse/SC-888) for this. We *can* drop the module.frequency attribute from our schema definitions if the frequency is PER_INSTANCE as fixup_module will properly set the default when "absent" to PER_INSTANCE anyway.

**Comment by TheRealFalcon** [comment]
Ooph, that's bad. We should add a test (not part of this PR) that we're not getting that warning anywhere. I may have changed that in previous modules such that modules are now running per-instance rather than what they're being declared to run as.

**Review by TheRealFalcon — APPROVED** [review]

**Pull request merged**